File: py/largest_palindrome_product.py
#!/usr/bin/env python

max_pal = 0
min_int = 100
max_int = 1000

def is_pal(num):
  num_str = str(num)
  for n in range(0,len(num_str)-1):
    if num_str[n] != num_str[len(num_str) -1 - n]:
      return False
  return True

for i in range(min_int,max_int):
  for j in range(min_int,max_int):
    num = i * j
    if is_pal(num) and num > max_pal:
      max_pal = num

print max_pal

File: ruby/largest_palindrome_product.rb
#!/usr/bin/env ruby


# Problem 4 from projecteuler.net
#
# A palindromic number reads the same both ways. The largest palindrome made
# from the product of two 2-digit numbers is 9009 = 91 × 99.
# 
# Find the largest palindrome made from the product of two 3-digit numbers.

def is_palindrome? num
  num = num.to_s
  num == num.reverse
end

max = 0

999.downto(100) do |i|
  999.downto(100) do |j|
    product = i * j
    if is_palindrome? product
      if product > max
        max = product
        break
      end
    end
  end
end
  
puts "max = #{max}"
# vim: ai ts=2 sts=2 et sw=2 ft=ruby
